README.md
# ac-regular

[![npm version](https://img.shields.io/npm/v/ac-regular.svg)](https://www.npmjs.com/package/ac-regular)
[![Build Status](https://img.shields.io/travis/tinper-bee/ac-regular/master.svg)](https://travis-ci.org/tinper-bee/ac-regular)
[![Coverage Status](https://coveralls.io/repos/github/tinper-bee/ac-regular/badge.svg?branch=master)](https://coveralls.io/github/tinper-bee/ac-regular?branch=master)
[![devDependency Status](https://img.shields.io/david/dev/tinper-bee/ac-regular.svg)](https://david-dm.org/tinper-bee/ac-regular#info=devDependencies)
[![NPM downloads](http://img.shields.io/npm/dm/ac-regular.svg?style=flat)](https://npmjs.org/package/ac-regular)
[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/tinper-bee/ac-regular.svg)](http://isitmaintained.com/project/tinper-bee/ac-regular "Average time to resolve an issue")
[![Percentage of issues still open](http://isitmaintained.com/badge/open/tinper-bee/ac-regular.svg)](http://isitmaintained.com/project/tinper-bee/ac-regular "Percentage of issues still open")


react ac-regular component for tinper-bee

some description...

## 依赖

- react >= 15.3.0
- react-dom >= 15.3.0
- prop-types >= 15.6.0

## 使用方法

```js
import Regular from 'ac-regular';
import 'ac-regular/build/regular.css'
```



## API

|参数|说明|类型|默认值|
|:--:|:--:|:--:|:--:|
|title|弹窗标题|string|正则表达式|
|regularTree|左侧树|array||
|backdrop|弹窗背景是否加深|Boolean|true|
|value|初始值|string|-|
|inputWidth|输入框宽度|number|240|
|onChange|点击确定的方法|function(value)|-|
|mode|返回数据类型|string|'java'/'js'|

### 左侧树数据要求
```json
{
    "regularTree":[
 
    {
        "name":"数学表达式校验",
        "code":"1",
        "id":"1",
        "child":[
            {"name":"至少n位的数字","code":"^\d{n,}$","id":"1-1","memo":"6位以上数字，^\d{6,}$"},
            {"name":"零和非零开头的数字","code":"^(0|[1-9][0-9]*)$","id":"1-2","memo":""},
            {"name":"非零开头的最多带两位小数的数字","code":"^([1-9][0-9]*)+(\.[0-9]{1,2})?$","id":"1-3","memo":""},
            {"name":"非零的负整数","code":"^-[1-9]\d*$","id":"1-4","memo":""},
            {"name":"浮点数","code":"^-?([1-9]\d*\.\d*|0\.\d*[1-9]\d*|0?\.0+|0)$","id":"1-5","memo":""},
          
        ]
    },
    {
        "name":"字符串校验",
        "code":"2",
        "id":"2",
        "child":[
            {"name":"汉字","code":"^[\u4e00-\u9fa5]{0,}$","id":"2-1","memo":""},
            {"name":"英文和数字","code":"^[A-Za-z0-9]+$","id":"2-2","memo":""},
            {"name":"数字、26个英文字母或下划线组成","code":"^\w+$","id":"2-3","memo":""},
            {"name":"中文、英文、数字包括下划线","code":"^[\u4E00-\u9FA5A-Za-z0-9_]+$","id":"2-4","memo":""},
            
        ]
    },
    {
        "name":"账号密码校验",
        "code":"3",
        "id":"3",
        "child":[
            {"name":"帐号","code":"^[a-zA-Z][a-zA-Z0-9_]{4,15}$","id":"3-1","memo":"字母开头，5-16字节，允许字母数字下划线"},
            {"name":"密码","code":"^[a-zA-Z]\w{5,17}$","id":"3-2","memo":"字母开头，长度6~18，含字母、数字和下划线"},
            {"name":"强密码","code":"^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[a-zA-Z0-9]{8,10}$","id":"2-3","memo":"大小写和数字组合，不使用特殊字符，长度 8~10"},
            {"name":"强密码","code":"^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,10}$","id":"2-4","memo":"含大小写和数字组合，可使用特殊字符，长度8-10"},
            
        ]
    },
    {
        "name":"联系方式校验",
        "code":"4",
        "id":"4",
        "child":[
            {"name":"手机号码","code":"^(13[0-9]|14[5|7]|15[0|1|2|3|4|5|6|7|8|9]|18[0|1|2|3|5|6|7|8|9])\d{8}$","id":"4-1","memo":"1568311xxxx"},
            {"name":"国内电话号码","code":"\d{3}-\d{8}|\d{4}-\d{7}","id":"4-2","memo":""},
            {"name":"电话号码","code":"((\d{11})|^((\d{7,8})|(\d{4}|\d{3})-(\d{7,8})|(\d{4}|\d{3})-(\d{7,8})-(\d{4}|\d{3}|\d{2}|\d{1})|(\d{7,8})-(\d{4}|\d{3}|\d{2}|\d{1}))$)","id":"4-3","memo":"支持手机号，3-4位区号，7-8位直播号码，1-4位分机号"},

        ]
    },
    {
        "name":"个人身份校验",
        "code":"5",
        "id":"5",
        "child":[
            {"name":"身份证号","code":"^[1-9]\d{5}(18|19|20|(3\d))\d{2}((0[1-9])|(1[0-2]))(([0-2][1-9])|10|20|30|31)\d{3}[0-9Xx]$","id":"5-1","memo":"(15位、18位数字)，最后一位是校验位，可能为数字或字符X"},
            {"name":"QQ号","code":"[1-9][0-9]{4,11}","id":"5-2","memo":"腾讯QQ号从10000开始"},
            {"name":"邮政编码","code":"[1-9]\d{5}(?!\d)","id":"5-3","memo":"中国邮政编码为6位数字"},
        ]
    },
    {
        "name":"网站信息校验",
        "code":"6",
        "id":"6",
        "child":[
            {"name":"Email地址","code":"^\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$","id":"6-1","memo":""},
            {"name":"域名","code":"[a-zA-Z0-9][-a-zA-Z0-9]{0,62}(\.[a-zA-Z0-9][-a-zA-Z0-9]{0,62})+\.?","id":"6-2","memo":""},
            {"name":"InternetURL","code":"((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)","id":"6-3","memo":""},
            {"name":"IP地址","code":"^((?:(?:25[0-5]|2[0-4]\\d|[01]?\\d?\\d)\\.){3}(?:25[0-5]|2[0-4]\\d|[01]?\\d?\\d))$","id":"6-4","memo":""},
        ]
    },
]
    
} 
```
#### 开发调试

```sh
$ npm install -g bee-tools
$ git clone https://github.com/tinper-bee/ac-regular
$ cd ac-regular
$ npm install
$ npm run dev
```


CHANGELOG.md
## [0.0.17](https://github.com/tinper-bee/ac-regular/compare/v0.0.16...v0.0.17) (2020-10-26)



## [0.0.16](https://github.com/tinper-bee/ac-regular/compare/v0.0.15...v0.0.16) (2020-09-24)



## [0.0.15](https://github.com/tinper-bee/ac-regular/compare/v0.0.14...v0.0.15) (2020-09-24)



## [0.0.14](https://github.com/tinper-bee/ac-regular/compare/v0.0.13...v0.0.14) (2020-09-22)



## [0.0.13](https://github.com/tinper-bee/ac-regular/compare/v0.0.12...v0.0.13) (2020-09-17)



## [0.0.12](https://github.com/tinper-bee/ac-regular/compare/v0.0.11...v0.0.12) (2020-09-15)



## [0.0.11](https://github.com/tinper-bee/ac-regular/compare/v0.0.10...v0.0.11) (2020-09-10)



## [0.0.10](https://github.com/tinper-bee/ac-regular/compare/v0.0.9...v0.0.10) (2020-09-10)



## [0.0.9](https://github.com/tinper-bee/ac-regular/compare/v0.0.8...v0.0.9) (2020-09-04)



## [0.0.8](https://github.com/tinper-bee/ac-regular/compare/v0.0.7...v0.0.8) (2020-09-01)



## [0.0.7](https://github.com/tinper-bee/ac-regular/compare/v0.0.6...v0.0.7) (2020-08-28)



## [0.0.6](https://github.com/tinper-bee/ac-regular/compare/v0.0.5...v0.0.6) (2020-08-28)



## [0.0.5](https://github.com/tinper-bee/ac-regular/compare/v0.0.4...v0.0.5) (2020-08-27)



## [0.0.4](https://github.com/tinper-bee/ac-regular/compare/v0.0.3...v0.0.4) (2020-08-27)



## [0.0.3](https://github.com/tinper-bee/ac-regular/compare/v0.0.2...v0.0.3) (2020-08-27)



## 0.0.2 (2020-08-27)





test/index.test.js
import React from 'react';
import {shallow, mount, render} from 'enzyme';
import {expect} from 'chai';
import Regular from '../src/index';

src/regular.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from 'bee-button';
import Modal from 'bee-modal';
import Icon from 'bee-icon';
import FormControl from 'bee-form-control';
import Left from './components/Left'
import Right from './components/Right'
import {regularTree} from './regularTree.js'


const propTypes = {};
const defaultProps = {};
class Regular extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showModal:false,
            title:props.title || '正则表达式',
            backdrop:props.backdrop || true,
            tree:props.regularTree ||regularTree,
            value:props.value || '',
            inputWidth:props.inputWidth || 240,
            inputValue:'',
            searchTree:[],
            mode:props.mode,
            memo:'',
            step:0,
            data:{}
        };
    }
    
    
    componentWillReceiveProps(porps){
        if(porps.value!=this.state.value){
            this.setState({value:porps.value || ''})
        }
       
    }
    close=()=>{
        this.setState({showModal:false,memo:''})
    }
    open=()=>{
        const {value}=this.state
        this.setState({showModal:true,inputValue:value})
    }
    submit=()=>{
        let {inputValue,mode}=this.state
        if(mode=='js'){
            inputValue = "/"+inputValue+"/"
        }
        this.setState({showModal:false,value:inputValue,memo:''})
        this.props.onChange&&this.props.onChange(inputValue)

    }
    valueChange=(data)=>{
        const memo = data.memo || ''
        this.inputChange(this.ch2Unicdoe(data.code))
        this.setState({memo:'实例说明：'+memo})
    }
    
    isChinese(s){
        return /[\u4e00-\u9fa5]/.test(s);
    }
    ch2Unicdoe=(str)=>{
        if(!str){
            return;
        }
        var unicode = '';
        for (var i = 0; i <  str.length; i++) {
            var temp = str.charAt(i);
            if(this.isChinese(temp)){
                unicode += '\\u' +  temp.charCodeAt(0).toString(16);
            }
            else{
                unicode += temp;
            }
        }
        return unicode;
    }
    inputChange=(inputValue,isStep,stepNow)=>{
        let {step,data} = this.state
        if(isStep!==true){
            step++
            data[step] = inputValue
        }else{
            step=stepNow
        }
        console.log(step,data)
        this.setState({inputValue,step,data})
    }
    onSearch=(searchValue)=>{
        const {tree=[]}=this.state
        let newTree = []
        tree.forEach((v)=>{
            v.child.forEach((v2)=>{
                if(v2.name.indexOf(searchValue)>-1){
                    newTree.push(v2)
                }
            })
        }) 
        this.setState({newTree,searchValue})
    }
    memoChange=(memo)=>{
       
        this.setState({memo})
    }
    render(){
        const {showModal,title,tree,value,inputWidth,memo,inputValue,newTree,searchValue,step,data} = this.state
        const {backdrop} =this.props
        return(
        <div className="ac-regular" style={{width:inputWidth}}>
            <div className='ac-regular-input' style={{width:inputWidth}}>
                <FormControl
                    className="demo1-input"
                    value={value}
                />
                <Icon type="uf-maxmize" onClick={this.open}/>
            </div>
           
            <Modal
                show={showModal}
                onHide={this.close}
                enforceFocus={false}
                backdrop={backdrop}
                className="ac-regular-modal"
            >
                <Modal.Header closeButton>
                <Modal.Title>{title}</Modal.Title>
                </Modal.Header>

                <Modal.Body>
                    <Left tree={tree} valueChange={this.valueChange} onSearch={this.onSearch} newTree={newTree} searchValue={searchValue} value={value}></Left>
                    <Right memo={memo} step={step} data={data} value={inputValue} inputChange={this.inputChange} memoChange={this.memoChange}></Right>
                </Modal.Body>

                <Modal.Footer className="text-center">
                    <Button colors="secondary" style={{ marginRight: 8 }}  onClick={this.close}>取消</Button>
                    <Button  colors="primary" className='submitBtn' onClick={this.submit}>确认</Button>
                </Modal.Footer>
            </Modal>
        </div>)
    }
};
Regular.propTypes = propTypes;
Regular.defaultProps = defaultProps;
export default Regular;

src/components/Left/index.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Select from 'bee-select';
import Icon from 'bee-icon';
import Pinyin from '../../ChinesePY'
import FormControl from 'bee-form-control';

const Option = Select.Option;
const propTypes = {};
const defaultProps = {};
class Left extends Component {
    constructor(props) {
        super(props);
        this.state = {
            selectValue:'0',
            activeIndex:-1,
            selectList:[],
            searchValue:'',
            treeAll:[]
        };
    }
    componentDidMount(){
        const {tree,value} = this.props
        // let selectList=[]
        // const newtree =JSON.parse(JSON.stringify(tree)) 
        // newtree.forEach((v)=>{
            
        //     v.id!='0'&&v.child.forEach((v2)=>{
        //         const newa = v2
        //         newa.name = v.name+'/'+newa.name
        //         selectList.push(newa)
        //     })
        // }) 
        // this.setState({selectList})
        let tree0=[] 
        let newobj={}
        let defualtIndex = -1
        tree[0].id!='0'?tree.forEach((v)=>{ 
            v.child.forEach((v2)=>{
                const id='0-'+v2.id
                newobj ={...v2,id} 
                tree0.push(newobj)
            })
            
        }):tree0=tree[0].child
        value&& tree0.forEach((v,i)=>{
            if(v.code == value){ this.valueChange(v,i) }
        })
        tree[0].id!='0'&&tree.unshift(
            {
                "name":"全部",
                "code":"0",
                "id":"0",
                "child":tree0
            })
        this.setState({treeAll:tree})
    }
    textChange=(selectValue)=>{
        const {value} = this.props
        const {treeAll} = this.state
        
        this.setState({selectValue,activeIndex:-1},()=>{
            // value&&treeAll[0].child.forEach((v,i)=>{
            //     if(v.code == value){ this.valueChange(v,i) }
            // })
        })
    }
    valueChange=(v,k)=>{
        this.props.valueChange(v)
        this.setState({activeIndex:k})
    }
    onSearchChange=(v)=>{
        const {selectValue,treeAll} = this.state
        const selectList = treeAll[selectValue].child
        if(v==''){
            this.setState({selectList,searchValue:''}) 
        }
    }
    onSearch = (searchValue) => {
        
        const {selectValue,treeAll} = this.state
        const list = treeAll[selectValue].child
        const selectList = list.filter((v)=>{
            return Pinyin.GetQP(v.name).indexOf( searchValue.toLowerCase() ) >= 0 ||v.name.indexOf(searchValue)>-1
        })
        // console.log(`selected ${value}`);
        // console.log(`selected item `,item);
        this.setState({searchValue,selectList})
        // this.valueChange(item,parseInt(item.id.split('-')[1])-1)
      };
    selectPinYin=(input,option)=>{
        if(input.charCodeAt()>=32&&input.charCodeAt()<=126){
            const value = option.props.children.toLowerCase();
            return Pinyin.GetQP(value).indexOf( input.toLowerCase() ) >= 0;
        }else{
            return option.props.children.toLowerCase().indexOf( input.toLowerCase() ) >= 0
        }
    }
    chineseChangePY=(item,input)=>{
        const qp = Pinyin.GetQP(item)
        const rgx = new RegExp(input, 'gi')
        return rgx.test(qp)
    }
    render(){
        const {selectList,activeIndex,selectValue,searchValue,treeAll} = this.state
       
        const loops =(data) => data.map((v,k)=>{
            return (
              <Option value={v.id} key={v.id}>{v.name}</Option>
            )
        })
        const loopsli =(data) => data.map((v,k)=>{
            return (
              <li className ={activeIndex==k?'regular-left-li active':'regular-left-li'} value={v.code} key={k} onClick={this.valueChange.bind(this,v,k)}>{v.name}</li>
            )
        })
        return(
        <div className="regular-left">
            <div className="regularleft-title">
                <span>正则表达式</span>
            </div>
            <Select 
                className ='regularleft-select'
                placeholder={''}
                style={{ width: 208}}
                value={selectValue}
                showSearch={true}
                optionFilterProp="children"
                onChange={this.textChange}>
                {loops(treeAll||[])}
             </Select>
            <FormControl
                className="regularleft-search"
                value={searchValue}
                style={{ width: 208}}
                onSearch={this.onSearch}
                onChange={this.onSearchChange}
                type="search"
            />
            {/* <Icon type="uf-search"/>
             <Select
                showSearch
               // combobox
                className="regularleft-search"
                style={{ width: 208 }}
                placeholder="搜索"
                optionFilterProp="children"
                filterOption={(input,Option)=>{return this.selectPinYin(input,Option)}}
                onChange={this.onSearch}
                dropdownClassName={'regularleft-search-drop'}
            >
                {
                selectList.map((da,i)=>{return (<Option key={i} title={da.name} value={da.code} item={da} >{da.name}</Option>)})
                }
            </Select> */}
           
             <ul  className ='regularleft-select-ul'>
                {
                    searchValue!=''?loopsli(selectList):treeAll[ parseInt(selectValue) ]?loopsli(treeAll[ parseInt(selectValue)].child):null
                }
             </ul>
        </div>)
    }
};
Left.propTypes = propTypes;
Left.defaultProps = defaultProps;
export default Left;

src/regularTree.js
export default {
    "regularTree":[
 
    {
        "name":"数学表达式校验",
        "code":"1",
        "id":"1",
        "child":[
            {"name":"至少n位的数字","code":"^\\d{n,}$","id":"1-1","memo":"6位以上数字，^\\d{6,}$"},
            {"name":"m-n位的数字","code":"^\\d{m,n}$","id":"1-2","memo":"3-8位的数字：^\\d{3,8}$"},
            {"name":"零和非零开头的数字","code":"^(0|[1-9][0-9]*)$","id":"1-3","memo":"零和非零开头的数字：^(0|[1-9][0-9]*)$"},
            {"name":"非零开头的最多带 n 位小数的数字","code":"^([1-9][0-9]*)+(\\.[0-9]{1,n})?$","id":"1-4","memo":"非零开头最多带 2 位小数的数字:^([1-9][0-9]*)+(\\.[0-9]{1,2})?$"},
            {"name":"非零的负整数","code":"^-[1-9]\\d*$","id":"1-5","memo":"小于零的负整数"},
            {"name":"非零的正整数","code":"^[1-9]\\d*$","id":"1-6","memo":"大于零的整数：^[1-9]\\d*$ 或 ^([1-9][0-9]*){1,3}$ 或 ^\\+?[1-9][0-9]*$"},
            {"name":"非负整数","code":"^\\d+$","id":"1-7","memo":"非负整数：^\\d+$ 或 ^[1-9]\\d*|0$"},
            {"name":"非正整数","code":"^-[1-9]\\d*|0","id":"1-8","memo":"非正整数：^-[1-9]\\d*|0$ 或 ^((-\\d+)|(0+))$"},
            {"name":"正浮点数","code":"^[1-9]\\d*\\.\\d*|0\\.\\d*[1-9]\\d*$","id":"1-9","memo":"正浮点数：^[1-9]\\d*\\.\\d*|0\\.\\d*[1-9]\\d*$"},
            {"name":"负浮点数","code":"^-([1-9]\\d*\\.\\d*|0\\.\\d*[1-9]\\d*)$","id":"1-10","memo":"负浮点数：^-([1-9]\\d*\\.\\d*|0\\.\\d*[1-9]\\d*)$ "},
            {"name":"浮点数","code":"^(-?\\d+)(\\.\\d+)?$ ","id":"1-11","memo":"浮点数：^(-?\\d+)(\\.\\d+)?$ "},
          
        ]
    },
    {
        "name":"字符串校验",
        "code":"2",
        "id":"2",
        "child":[
            {"name":"中文字符","code":"^[\u4e00-\u9fa5]{0,}$","id":"2-1","memo":"2-6位中文名：^[\\u4e00-\\u9fa5]{2,6}$"},
            {"name":"长度为3-20的字符串","code":"^.{3,20}$","id":"2-2","memo":"长度为3-20的字符串：^.{3,20}$"},
            {"name":"由26个英文字母","code":"^[A-Za-z]+$","id":"2-3","memo":"由26个英文字母组成的字符串：^[A-Za-z]+$"},
            {"name":"26个英文字母和数字","code":"^[A-Za-z0-9]+$","id":"2-4","memo":"26个英文字母和数字组成的字符串：^[A-Za-z0-9]+$"},
            {"name":"数字、26个英文字母或下划线","code":"^\\w+$","id":"2-5","memo":"由数字、26个英文字母或者下划线组成的字符串：^\\w+$ 或 ^\\w{3,20}$"},
            {"name":"中文、英文、数字包括下划线","code":"^[\u4E00-\u9FA5A-Za-z0-9_]+$","id":"2-6","memo":"中文、英文、数字包括下划线组成字符串：^[\\u4E00-\\u9FA5A-Za-z0-9_]+$"},
            {"name":"中文、英文、数字但不包括下划线等","code":"^[\u4E00-\u9FA5A-Za-z0-9]+$ ","id":"2-7","memo":"中文、英文、数字但不包括下划线等组成的字符串：^[\\u4E00-\\u9FA5A-Za-z0-9]+$ "},
            
        ]
    },
    {
        "name":"账号密码校验",
        "code":"3",
        "id":"3",
        "child":[
            {"name":"帐号","code":"^[a-zA-Z][a-zA-Z0-9_]{4,15}$","id":"3-1","memo":"帐号(字母开头，5-16字节，允许字母数字下划线)：^[a-zA-Z][a-zA-Z0-9_]{4,15}$"},
            {"name":"密码","code":"^[a-zA-Z]\\w{5,17}$","id":"3-2","memo":"密码(字母开头，长度6~18，含字母、数字和下划线)：^[a-zA-Z]\\w{5,17}$"},
            {"name":"强密码","code":"^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])[a-zA-Z0-9]{8,10}$","id":"3-3","memo":"强密码(大小写和数字组合，不使用特殊字符，长度 8~10 )：^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])[a-zA-Z0-9]{8,10}$"},
            {"name":"强密码","code":"^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{8,10}$","id":"3-4","memo":"强密码(含大小写和数字组合，可使用特殊字符，长度8-10)：^(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{8,10}$"},
            
        ]
    },
    {
        "name":"联系方式校验",
        "code":"4",
        "id":"4",
        "child":[
            {"name":"手机号码","code":"^(13[0-9]|14[5|7]|15[0-9]|18[0-9])\\d{8}$","id":"4-1","memo":"手机号码（中国大陆）：^(13[0-9]|14[5|7]|15[0-9]|18[0-9])\\d{8}$，例如 136xxxxxxxx"},
            {"name":"电话号码","code":"^(\\(\\d{3,4}-)|\\d{3.4}-)?\\d{7,8}$","id":"4-2","memo":"匹配形式如 ('XXX-XXXXXXX'、'XXXX-XXXXXXX'、'XXX-XXXXXXXX'、'XXXX-XXXXXXXX'、'XXXXXXX'和'XXXXXXXX)"},
            {"name":"固定电话号码","code":"\\d{3}-\\d{8}|\\d{4}-\\d{7}","id":"4-3","memo":"固定电话号码（中国大陆）:\\d{3}-\\d{8}|\\d{4}-\\d{7} , 匹配形式如 0511-4405222 或 021-87888822"},

        ]
    },
    {
        "name":"个人身份校验",
        "code":"5",
        "id":"5",
        "child":[
            {"name":"身份证号","code":"(^\\d{15}$)|(^\\d{18}$)|(^\\d{17}(\\d|X|x)$)","id":"5-1","memo":"身份证号(15位、18位数字)，最后一位是校验位，可以为X"},
            {"name":"QQ号","code":"[1-9][0-9]{4,11}","id":"5-2","memo":"5-10位QQ号码 [1-9][0-9]{4,11}"},
            {"name":"邮政编码","code":"^[1-9]\\d{5}(?!\\d)","id":"5-3","memo":"6位邮政编码 ^[1-9]\\d{5}(?!\\d)"},
            {"name":"Email地址","code":"^[A-Za-z0-9-._]+@[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,6})$","id":"5-4","memo":"邮箱地址 ^[A-Za-z0-9-._]+@[A-Za-z0-9-]+(\.[A-Za-z0-9]+)*(\.[A-Za-z]{2,6})$"},
        ]
    },
    {
        "name":"网站信息校验",
        "code":"6",
        "id":"6",
        "child":[
            {"name":"IPv4地址","code":"((2(5[0-5]|[0-4]\\d))|[0-1]?\\d{1,2})(\\.((2(5[0-5]|[0-4]\\d))|[0-1]?\\d{1,2})){3}","id":"6-1","memo":"IPv4地址的匹配形式如下：192.168.56.1"},
            {"name":"域名","code":"[a-zA-Z0-9][-a-zA-Z0-9]{0,62}(\\.[a-zA-Z0-9][-a-zA-Z0-9]{0,62})+\\.?","id":"6-2","memo":"域名的匹配形式如下：yonyou.com"},
            {"name":"IP地址","code":"/^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/","id":"6-3","memo":"IP地址：/^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)、\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/"},
            {"name":"InternetURL","code":"((([A-Za-z]{3,9}:(?:\\/\\/)?)(?:[-;:&=\\+\\$,\\w]+@)?[A-Za-z0-9.-]+(:[0-9]+)?|(?:ww‌​w.|[-;:&=\\+\\$,\\w]+@)[A-Za-z0-9.-]+)((?:\\/[\\+~%\\/.\\w-_]*)?\\??(?:[-\\+=&;%@.\\w_]*)#?‌​(?:[\\w]*))?)","id":"6-4","memo":"URL地址"},
        ]
    },
]
    
} 